src/main/java/uk/gov/hmcts/ccd/domain/service/search/CaseSearchResultGenerator.java

```diff
@@ -0,0 +1,177 @@
+package uk.gov.hmcts.ccd.domain.service.search;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
+import uk.gov.hmcts.ccd.data.user.UserRepository;
+import uk.gov.hmcts.ccd.domain.model.aggregated.CommonField;
+import uk.gov.hmcts.ccd.domain.model.definition.*;
+import uk.gov.hmcts.ccd.domain.model.search.CaseSearchResult;
+import uk.gov.hmcts.ccd.domain.model.search.elasticsearch.*;
+import uk.gov.hmcts.ccd.domain.service.aggregated.AuthorisedGetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.GetCaseTypeOperation;
+import uk.gov.hmcts.ccd.domain.service.aggregated.SearchQueryOperation;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadRequestException;
+import uk.gov.hmcts.ccd.endpoint.exceptions.BadSearchRequest;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+
+@Service
+public class CaseSearchResultGenerator {
+
+    private final UserRepository userRepository;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final SearchQueryOperation searchQueryOperation;
+
+    public CaseSearchResultGenerator(@Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                                     @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) final GetCaseTypeOperation getCaseTypeOperation,
+                                     final SearchQueryOperation searchQueryOperation) {
+        this.userRepository = userRepository;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.searchQueryOperation = searchQueryOperation;
+    }
+
+    public UICaseSearchResult execute(final String caseTypeId,
+                                      final CaseSearchResult caseSearchResult,
+                                      final String useCase) {
+        // TODO: Filter out fields from result that haven't been requested before returning (RDM-8556)
+        return new UICaseSearchResult(
+            buildHeaders(caseTypeId, useCase, caseSearchResult),
+            buildItems(useCase, caseSearchResult),
+            caseSearchResult.getTotal(),
+            useCase
```

Comment:
minor: sorry Dan I changed my mind only if it's not a big change can we remove use case? I should have applied YAGNI